### PR TITLE
Vlan2

### PR DESCRIPTION
**Pull request by xnox**

### CODE REVIEW

**Comment by mwhudson** [comment]
I updated my branch a bunch and rebased this one on top of it.

File: subiquitycore/controllers/network.py [hunk 1]
```diff
@@ -237,6 +238,21 @@ def netplan_path(self):
             netplan_config_file_name = '00-snapd-config.yaml'
         return os.path.join(self.root, 'etc/netplan', netplan_config_file_name)
 
+    def add_vlan(self, device, vlan):
```
Comment (mwhudson) [inline subiquitycore/controllers/network.py:241]: I'd like to think of a way to do this where dry-run doesn't have to run as root. But this is probably OK for this branch.

File: subiquitycore/models/network.py [hunk 2]
```diff
@@ -45,6 +46,11 @@ class Networkdev:
     def __init__(self, net_info, configuration):
         self._net_info = net_info
         self._configuration = configuration
+        if self.type == 'vlan':
+            # probably should parse /proc/self/net/vlan/config instead
```
Comment (mwhudson) [inline subiquitycore/models/network.py:50]: Weeellll reallllly probert's _rtnetlink stuff should be looking out for this?

File: subiquitycore/models/network.py [hunk 3]
```diff
@@ -23,6 +23,7 @@
 
 NETDEV_IGNORED_IFACE_NAMES = ['lo']
 NETDEV_IGNORED_IFACE_TYPES = ['bridge', 'tun', 'tap', 'dummy', 'sit']
+NETDEV_WHITELIST_IFACE_TYPES = ['vlan']
```
Comment (mwhudson) [inline subiquitycore/models/network.py:26]: I think the name of this list should have VIRTUAL in it somewhere?

File: subiquitycore/ui/views/network.py [hunk 4]
```diff
@@ -220,9 +234,17 @@ def _build_model_inputs(self):
...
-        if isinstance(self._w, StretchyOverlay) and \
-           hasattr(self._w.stretchy, 'refresh_model_inputs'):
+        if isinstance(self._w, StretchyOverlay) and hasattr(
+                self._w.stretchy, 'refresh_model_inputs'):
             self._w.stretchy.refresh_model_inputs()
+        # we have heading, and then three lines per interface
+        # selectable line, extra line, whitespace line
+        # and focus ends up on the last whitespace line
+        # despite it, not being selectable. *derp*
+        current_focus = self.device_table.focus_position
```
Comment (mwhudson) [inline subiquitycore/ui/views/network.py:244]: argh we need to do something better for this (and on filesystem screen too)

probably not on this branch though!

**Review by mwhudson — APPROVED** [review]
I have a few comments but nothing major, I think we should merge and iterate tbh.

File: subiquitycore/models/network.py (hunk 2 above)
Comment (xnox) [reply subiquitycore/models/network.py:50]: Quite.... but it doesn't do that now =/

Comment (xnox) [reply subiquitycore/models/network.py:50]: https://trello.com/c/ayU1R0Al

File: subiquitycore/controllers/network.py (hunk 1 above)
Comment (xnox) [reply subiquitycore/controllers/network.py:241]: https://trello.com/c/2Opt8urP

File: subiquitycore/models/network.py (hunk 3 above)
Comment (xnox) [reply subiquitycore/models/network.py:26]: https://trello.com/c/8FdcEE3N

File: subiquitycore/ui/views/network.py (hunk 4 above)
Comment (xnox) [reply subiquitycore/ui/views/network.py:244]: https://trello.com/c/qrMBy4Ve

**Comment by xnox** [comment]
Cards created for all the comments.
Rebased.
Merging after it will pass CI.

**Pull request closed without merging**